ta/pkcs11/src/processing_symm.c

```diff
@@ -461,6 +533,31 @@ init_tee_operation(struct pkcs11_session *session,
 		if (proc_params->size)
 			return PKCS11_CKR_MECHANISM_PARAM_INVALID;
 
+		TEE_MACInit(session->processing->tee_op_handle, NULL, 0);
+		rc = PKCS11_CKR_OK;
+		break;
+	case PKCS11_CKM_MD5_HMAC_GENERAL:
+	case PKCS11_CKM_SHA_1_HMAC_GENERAL:
+	case PKCS11_CKM_SHA224_HMAC_GENERAL:
+	case PKCS11_CKM_SHA256_HMAC_GENERAL:
+	case PKCS11_CKM_SHA384_HMAC_GENERAL:
+	case PKCS11_CKM_SHA512_HMAC_GENERAL:
+		if (proc_params->size != sizeof(uint32_t))
+			return PKCS11_CKR_MECHANISM_PARAM_INVALID;
+
+		pkcs11_data = TEE_Malloc(sizeof(uint32_t),
+					 TEE_MALLOC_FILL_ZERO);
+		if (!pkcs11_data)
+			return PKCS11_CKR_DEVICE_MEMORY;
+
+		memcpy(pkcs11_data, proc_params->data, sizeof(uint32_t));
+
+		rc = input_hmac_len_is_valid(proc_params, *pkcs11_data);
+		if (rc)
+			return rc;
```

Comment:
we leak memory here (pkcs11_data) if the len is not valid.

Reply:
Good catch! Thanks!